### PR TITLE
Updated USER_GUIDE regarding deprecation

### PR DESCRIPTION
**Pull request by saimedhi**
### Description
Updated USER_GUIDE regarding deprecation

By submitting this pull request, I confirm that my contribution is made under the terms of the Apache 2.0 license.
For more information on following the Developer Certificate of Origin and signing off your commits, please check [here](https://github.com/opensearch-project/opensearch-dsl-py/blob/main/CONTRIBUTING.md#developer-certificate-of-origin).

### CODE REVIEW

File: USER_GUIDE.md [hunk 1]
```diff
@@ -3,6 +3,8 @@
   - [Sample code](#sample-code)
 # User Guide
 
+The OpenSearch high-level Python client (opensearch-dsl-py) will be deprecated with its final release version as 2.1.0. We recommend switching to the OpenSearch Python client (opensearch-py) instead. {: .warning}
```
Comment (VachaShah) [inline USER_GUIDE.md:6]: Lets add this as a banner on the README so it appears on the Github and PYPI landing page.

Comment (saimedhi) [reply USER_GUIDE.md:6]: ok

Comment (wbeckler) [reply USER_GUIDE.md:6]: Maybe mention in the same places that the functionality was moved to the python client? Like "We recommend switching to the OpenSearch Python client (opensearch-py) which now includes the functionality of opensearch-py-dsl"

Comment (saimedhi) [reply USER_GUIDE.md:6]: ok

**Pull request closed without merging**